test/unit/asprintf.c

```diff
@@ -17,11 +17,11 @@
 
 Test(asprintf, valid) {
     assert_cr_asprintf_format_is_valid("%c", 'a');
-    assert_cr_asprintf_format_is_valid("%hd", 42);
+    assert_cr_asprintf_format_is_valid("%hd", (short) 42);
```

Comment:
I'm surprised gcc even cares given that any value shorter than int gets implicitly passed as an int through varargs. I guess the warning is there just for correctness.

Reply:
I was surprised too. :)